tests/test_client_session.py

```diff
@@ -467,6 +467,60 @@ async def create_connection(req, traces, timeout):
         c.__del__()
 
 
+@pytest.mark.parametrize("protocol", ["http", "https", "ws", "wss"])
+async def test_ws_connect_allowed_protocols(
+    create_session: Any,
+    create_mocked_conn: Any,
+    protocol: str,
+    ws_key: Any,
+    key_data: Any,
+) -> None:
+    url = URL(f"{protocol}://example.com")
+    req = mock.Mock()
```

Comment:
For all these things, could we use create_autospec() (ideally with spec_set=True)? Should be able to avoid pretty much all direct Mock/AsyncMock() creations with that.

Reply:
68ea65fed759e6f8ae69e7146127a30e26e9552f

spec_set=True worked for `ClientRequest` but not for `ClientResponse` since it wouldn't accept the cookies when I did that.